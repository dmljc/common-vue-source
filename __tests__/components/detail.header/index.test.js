/**
 * @jest-environment jsdom
 */
import { createLocalVue, mount } from '@vue/test-utils';
import Compose from '@/main';
import DetailHeader from '@/components/detail.header';

let wrapper;
const localVue = createLocalVue();
localVue.use(Compose);
const customMount = options => mount(DetailHeader, Object.assign({localVue}, options));

describe('detail-header default props & data', () => {
    it('detail-header default props & data', () => {
        wrapper = customMount();
        expect(wrapper.element).toMatchSnapshot();
    });
});

describe('detail-header customs props', () => {
    it('detail-header customs props case1', () => {
        wrapper = customMount({
            propsData: {
                header: '标题',
                status: '状态'
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('detail-header customs props case2', () => {
        wrapper = customMount({
            propsData: {
                header: '标题',
                subHeader: '子标题',
                status: '状态'
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('detail-header customs props case3', () => {
        wrapper = customMount({
            propsData: {
                header: '标题',
                subHeader: '子标题',
                status: '状态'
            },
            slots: {
                operation: '<el-button title="导出" />'
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});