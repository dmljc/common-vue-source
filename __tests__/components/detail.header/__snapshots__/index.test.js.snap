// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detail-header customs props detail-header customs props case1 1`] = `
<div
  class="header"
>
  <div
    class="el-card is-always-shadow"
  >
    <!---->
    <div
      class="el-card__body"
    >
      <div
        class="header-title"
      >
        <div
          class="el-row"
        >
          <div
            class="el-col el-col-16"
          >
            <span>
              标题
            </span>
          </div>
           
          <!---->
           
          <div
            align="right"
            class="header__opts el-col el-col-8"
          />
           
          <div
            class="el-col el-col-24"
          >
            <span
              class="header__status f-mt10"
            >
              状态
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`detail-header customs props detail-header customs props case2 1`] = `
<div
  class="header"
>
  <div
    class="el-card is-always-shadow"
  >
    <!---->
    <div
      class="el-card__body"
    >
      <div
        class="header-title"
      >
        <div
          class="el-row"
        >
          <div
            class="el-col el-col-8"
          >
            <span>
              标题
            </span>
          </div>
           
          <div
            class="el-col el-col-8"
          >
            <span>
              子标题
            </span>
          </div>
           
          <div
            align="right"
            class="header__opts el-col el-col-8"
          />
           
          <div
            class="el-col el-col-24"
          >
            <span
              class="header__status f-mt10"
            >
              状态
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`detail-header customs props detail-header customs props case3 1`] = `
<div
  class="header"
>
  <div
    class="el-card is-always-shadow"
  >
    <!---->
    <div
      class="el-card__body"
    >
      <div
        class="header-title"
      >
        <div
          class="el-row"
        >
          <div
            class="el-col el-col-8"
          >
            <span>
              标题
            </span>
          </div>
           
          <div
            class="el-col el-col-8"
          >
            <span>
              子标题
            </span>
          </div>
           
          <div
            align="right"
            class="header__opts el-col el-col-8"
          >
            <button
              class="el-button el-button--default"
              title="导出"
              type="button"
            >
              <!---->
              <!---->
              <!---->
            </button>
          </div>
           
          <div
            class="el-col el-col-24"
          >
            <span
              class="header__status f-mt10"
            >
              状态
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`detail-header default props & data detail-header default props & data 1`] = `
<div
  class="header"
>
  <div
    class="el-card is-always-shadow"
  >
    <!---->
    <div
      class="el-card__body"
    >
      <div
        class="header-title"
      >
        <div
          class="el-row"
        >
          <div
            class="el-col el-col-16"
          >
            <span>
              
            </span>
          </div>
           
          <!---->
           
          <div
            align="right"
            class="header__opts el-col el-col-8"
          />
           
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;
